deployment/ccip/changeset/cs_update_rmn_config.go

```diff
@@ -0,0 +1,256 @@
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	mcmsWrappers "github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_home"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_remote"
+)
+
+type SetRMNHomeCandidateConfig struct {
+	HomeChainSelector uint64
+	RMNStaticConfig   rmn_home.RMNHomeStaticConfig
+	RMNDynamicConfig  rmn_home.RMNHomeDynamicConfig
+	DigestToOverride  [32]byte
+}
+
+func (c SetRMNHomeCandidateConfig) Validate() error {
+	if c.HomeChainSelector == 0 {
+		return fmt.Errorf("HomeChainSelector must be set")
+	}
+	if len(c.RMNDynamicConfig.OffchainConfig) != 0 {
+		return fmt.Errorf("RMNDynamicConfig.OffchainConfig must be empty")
+	}
+	if len(c.RMNStaticConfig.OffchainConfig) != 0 {
+		return fmt.Errorf("RMNStaticConfig.OffchainConfig must be empty")
+	}
+
+	return nil
+}
+
+type PromoteRMNHomeCandidateConfig struct {
+	HomeChainSelector uint64
+}
+
+func (c PromoteRMNHomeCandidateConfig) Validate() error {
+	if c.HomeChainSelector == 0 {
+		return fmt.Errorf("HomeChainSelector must be set")
+	}
+
+	return nil
+}
+
+func NewSetRMNHomeCandidateConfigChangeset(e deployment.Environment, config SetRMNHomeCandidateConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(e)
+	lggr := e.Logger
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	rmnHome := state.Chains[config.HomeChainSelector].RMNHome
+
+	setCandidateTx, err := rmnHome.SetCandidate(deployment.SimTransactOpts(), config.RMNStaticConfig, config.RMNDynamicConfig, config.DigestToOverride)
+	if err != nil {
+		lggr.Errorw("Failed to build call data to set RMNHome candidate digest", "err", err)
+		return deployment.ChangesetOutput{}, err
+	}
+
+	op := mcms.Operation{
+		To:    rmnHome.Address(),
+		Data:  setCandidateTx.Data(),
+		Value: big.NewInt(0),
+	}
+
+	prop, err := buildProposal(e, op, state, config.HomeChainSelector)
+
+	return deployment.ChangesetOutput{
+		Proposals: []timelock.MCMSWithTimelockProposal{*prop},
+	}, nil
+}
+
+func NewPromoteCandidateConfigChangeset(e deployment.Environment, config PromoteRMNHomeCandidateConfig) (deployment.ChangesetOutput, error) {
+	state, err := LoadOnchainState(e)
+	lggr := e.Logger
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to load onchain state: %w", err)
+	}
+	rmnHome := state.Chains[config.HomeChainSelector].RMNHome
+
+	currentCandidateDigest, err := rmnHome.GetCandidateDigest(nil)
+	if err != nil {
+		lggr.Errorw("Failed to get RMNHome candidate digest", "err", err)
```

Comment:
please include chain name in all logging after this [PR](https://github.com/smartcontractkit/chainlink/pull/15535) is merged